### PR TITLE
Sniper Rifle - Features Zoom

### PR DESCRIPTION
**Pull request by Cheridan**
If I didn't go to sleep, it still counts as the 19th day.
Late entry due to personal reasons.

Ordered by Kevinz000!
Bay-style sniper weapon with zoom minus snowflake code
Other features coming soon :tm: 

Please test to make sure stupid stuff doesn't happen

### CODE REVIEW

**Comment by Nienhaus** [comment]
So you moves the camera forward like the bay one?

**Comment by GunHog** [comment]
Origin Tech is Combat 6...this is going to be a Wizard/Admin only weapon, huh?

**Comment by Fox-McCloud** [comment]
:+1: on the idea :-1: on implementation.

https://github.com/Baystation12/Baystation12/pull/6588

Bay handles this at the item level now so that any gun can be zoomed in; it also means you can make items like binoculars and such.

**Comment by GunHog** [comment]
So we could...MAKE SCOPES...

**Comment by Fox-McCloud** [comment]
@GunHog 

Exactly; you could have a var for certain guns that allows for scopes to be attachable or somethin' er other---then just add in stand-alone attachable scopes.

**Comment by kevinz000** [comment]
Gimmicks are cool <3

**Comment by Cheridan** [comment]
I'm not gonna add vars to every item that are going to be used in one, maybe two.

**Comment by kevinz000** [comment]
Can't we port baystation's insta-beams?

**Comment by Fox-McCloud** [comment]
@kevinz000 

Because hit-scan lasers and projectiles are so much fun to play against.

Bay's lasers have a sizeable firing delay on them and they do a massive 40 damage per shot; I don't think they'd be a good idea on a server that has a large amount of gameplay.

**Comment by kevinz000** [comment]
Even if we onlny limit it to things like the sniper and remove the damage, just makes instastun? :)

**Comment by Cheridan** [comment]
Yeah, the plan is still to add either baystation beams or greatly sped-up shots. I just have to get around to making them work with our drastically different guncode.

**Comment by Fox-McCloud** [comment]
@Cheridan 

I'd just keep projectile beams the way they are now, personally--just allow for diagonal shots. That's pretty much the system Goon has (and Bay, too, only their beams are instant); it still allows for clutch dodges without shots looking terrible.

**Comment by RemieRichards** [comment]
if you mean Bay's beams that are long and produce a considerable well "beam", those got cleaned up to be tracer and muzzle effects in Loganbacca's great rewrite.

With that rewrite the difference between their hitscan and our non histcan weapons is literally a ```hitscan``` variable.

**Comment by Fox-McCloud** [comment]
Ahh, fair enough, though I don't personally think hitscan is necessarily a good thing on a server like TG--though, again, you could easily disable that if it's literally 1 var.

File: code/modules/projectiles/guns/energy/special.dm [hunk 1]
```diff
@@ -328,3 +328,82 @@
...
+	if(istype(user,/mob/living/carbon/human))
+		var/mob/living/carbon/human/H = user
+		H.unzoom()
+	..()
+
+/obj/item/weapon/gun/energy/sniper/proc/zoom(var/mob/living/carbon/human/user, var/tileoffset = 11,var/viewsize = 12) //tileoffset is client view offset in the direction the user is facing. viewsize is how far out this thing zooms. 7 is normal view
+	var/cannotzoom
+
+	if(user.stat)
+		usr << "You can't focus your eyes while you're dead."
```
Comment (Giacom) [inline code/modules/projectiles/guns/energy/special.dm]: You should use "user" instead of "usr". You should also use span classes.

**Comment by Xhuis** [comment]
'the [src]' would probably make it end up saying "the the"

File: code/modules/projectiles/guns/energy/special.dm [hunk 2]
```diff
@@ -328,3 +328,82 @@
...
+	slot_flags = SLOT_BACK
+	fire_delay = 35
+	w_class = 4.0
+	var/zoomed = 0
+
+/obj/item/weapon/gun/energy/sniper/attack_self(mob/living/user as mob)
+	if(istype(user,/mob/living/carbon/human))
+		zoom(user)
+	else
+		usr << "You are unable to focus through the [src]."
```
Comment (Xhuis) [inline code/modules/projectiles/guns/energy/special.dm]: The [src] probably leads to it saying 'the the'

Comment (Dorsisdwarf) [inline code/modules/projectiles/guns/energy/special.dm]: Monkies cannot snipe their masters 0/10

**Comment by Razharas** [comment]
Is this alive?
Is cheri alive?

**Comment by Cheridan** [comment]
yes i'm just busy

**Comment by Razharas** [comment]
Open this when you are ready to finish it

**Pull request closed without merging**